src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorizeSearchIntegrationTest.java

```diff
@@ -0,0 +1,820 @@
+
+@QuarkusIntegrationTest
+@QuarkusTestResource(DseTestResource.class)
+@TestClassOrder(ClassOrderer.OrderAnnotation.class)
+public class VectorizeSearchIntegrationTest extends AbstractNamespaceIntegrationTestBase {
+
+  private static final String collectionName = "my_collection_vectorize";
+
+  @Nested
+  @Order(1)
+  class CreateCollection {
+    @Test
+    public void happyPathVectorSearch() {
+      String json =
+          """
+                {
+                  "createCollection": {
+                    "name" : "my_collection_vectorize",
+                    "options": {
+                      "vector": {
+                        "size": 5,
+                        "function": "cosine"
+                      },
+                      "vectorize": {
+                        "service" : "custom",
+                        "options" : {
+                          "modelName": "text-embedding-ada-002"
+                        }
+                      }
+                    }
+                  }
+                }
+                """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+    }
+  }
+
+  @Nested
+  @Order(2)
+  class InsertOneCollection {
+    @Test
+    public void insertVectorSearch() {
+      String json =
+          """
+            {
+               "insertOne": {
+                  "document": {
+                      "_id": "1",
+                      "name": "Coded Cleats",
+                      "description": "ChatGPT integrated sneakers that talk to you",
+                      "$vectorize": "ChatGPT integrated sneakers that talk to you"
+                  }
+               }
+            }
+            """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.insertedIds[0]", is("1"))
+          .body("data", is(nullValue()))
+          .body("errors", is(nullValue()));
+
+      json =
+          """
+            {
+              "find": {
+                "filter" : {"_id" : "1"}
+              }
+            }
+            """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents[0]._id", is("1"))
+          .body("data.documents[0].$vector", is(notNullValue()))
```

Comment:
Should probably also check that we get an array of floats (with known size but no need to match values)